templates/onetimesecret-deployment.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "onetimesecret-chart.onetimesecret.fullname" . }}
+  labels:
+    app: {{ include "onetimesecret-chart.onetimesecret.name" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ include "onetimesecret-chart.onetimesecret.name" . }}
+  template:
+    metadata:
+      labels:
+        app: {{ include "onetimesecret-chart.onetimesecret.name" . }}
+    spec:
+      containers:
+      - name: {{ include "onetimesecret-chart.onetimesecret.name" . }}
+        image: "{{ .Values.onetimesecret.image.repository }}:{{ include "onetimesecret-chart.image.tag" . }}"
+        ports:
+        - containerPort: 3000
+        env:
```

Comment:
So looking at the variables I think a better approach that I have seen is doing conditionals so that when something is defined it gets used otherwise it gets defaulted. Here is an example of what that might look like:

https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-applicationset/deployment.yaml#L54-L56

In the case of REDIS_URL, if redis.enabled is false then it should probably take the value from the user as is.

In the case of SMTP host i'm thinking something similar. If it's defined take it from the user otherwise what are sane defaults to use? Is there a way to disable SMTP entirely or should the app error out? Is it required?

Let me know your thoughts. If you have seen a better practice i'm all for it. As for where the values should be commented out, I think having them in the values.yaml would be helpful so that the cosumer doesn't have to dig through the templates